6_nodejs/week3/app.js

```diff
@@ -0,0 +1,50 @@
+import knex from 'knex'
+const knexInstance = knex({
+  client: "mysql2",
+  connection: {
+    host: process.env.DB_HOST || "127.0.0.1",
+    port: process.env.DB_PORT || 3306,
+    user: process.env.DB_USER || "root",
+    password: process.env.DB_PASSWORD || "my-secret-pw",
+    database: process.env.DB_NAME || "hyf_node_week3_warmup",
+    multipleStatements: true,
+  },
+});
+
+import express from 'express'
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+contactsAPIRouter.get("/", async (req, res) => {
+  let query = knexInstance.select("*").from("contacts");
+
+  if ("sort" in req.query) {
+    const orderBy = req.query.sort.toString();
+    if (orderBy.length > 0) {
+        // Use raw expression for proper bindings
+        query = query.orderByRaw(knex.raw('?', [orderBy]));
```

Comment:
This fails on my machine as `raw` is not recognized. It seems `knexInstance.raw` exists though, but I am unsure if this behaves as intended. [This page](https://knexjs.org/guide/raw.html#raw-parameter-binding) might be helpful for using `raw`. 😄

You might need some additional logic to handle the keywords `ASC` and `DESC`.